frontend/src/components/FormElements/TimeField.jsx

```diff
@@ -0,0 +1,175 @@
+import { useState, useRef } from "react";
+import TextField from "@mui/material/TextField";
+import { Controller } from "react-hook-form";
+import { Grid, Box, Typography, FormHelperText } from "@mui/material";
+
+const TimeField = ({ label, name, control, rules }) => {
+  const inputRefs = useRef([null, null, null]);
+
+  return (
+    <Controller
+      name={name}
+      control={control}
+      rules={rules}
+      render={({ field: { onChange, value }, fieldState: { error } }) => {
+        const [time, setTime] = useState({
+          minutes: "",
+          seconds: "",
+          milliseconds: "",
+        });
+
+        const handleChange = (field, newValue, currentIndex) => {
+          const trimmedValue = newValue.slice(0, 2);
+          setTime((prevTime) => {
+            const updatedTime = { ...prevTime, [field]: trimmedValue };
+            const formattedTime = `${
+              updatedTime.minutes.padStart(2, "0") || "00"
+            }:${updatedTime.seconds.padStart(2, "0") || "00"}.${
+              updatedTime.milliseconds.padStart(2, "0") || "00"
+            }`;
+            if (formattedTime === "00:00.00") {
+              onChange(null);
+            } else {
+              onChange(formattedTime);
+            }
+            if (
+              trimmedValue.length === 2 &&
+              currentIndex < inputRefs.current.length - 1
+            ) {
+              inputRefs.current[currentIndex + 1]?.focus();
+            }
+            return updatedTime;
+          });
+        };
+
+        const handleKeyDown = (e, field, currentIndex) => {
+          if (e.key === "Backspace" && !time[field]) {
+            if (currentIndex > 0) {
+              inputRefs.current[currentIndex - 1]?.focus();
+            }
+          } else if (e.key === "ArrowLeft" && currentIndex > 0) {
+            inputRefs.current[currentIndex - 1]?.focus();
+          } else if (
+            e.key === "ArrowRight" &&
+            currentIndex < inputRefs.current.length - 1
+          ) {
+            inputRefs.current[currentIndex + 1]?.focus();
+          }
+        };
+
+        const handlePaste = (e) => {
+          e.preventDefault();
+          const pastedText = e.clipboardData
+            .getData("text/plain")
+            .replace(/\D/g, "");
+          if (pastedText.length === 6) {
+            setTime({
+              minutes: pastedText.slice(0, 2),
+              seconds: pastedText.slice(2, 4),
+              milliseconds: pastedText.slice(4, 6),
+            });
+            onChange(
+              `${pastedText.slice(0, 2)}:${pastedText.slice(
+                2,
+                4
+              )}.${pastedText.slice(4, 6)}`
+            );
+            inputRefs.current[2]?.focus();
+          }
+        };
```

Comment:
We could improve this function for allowing the user paste values with length < 6 depending in which field is trying to paste the value. This could be a future improvement, the paste functionality is working right now for a value with 6 numeric digits in it.